docs/standard/serialization/binaryformatter-migration-guide/migrate-to-system-text-json.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "BinaryFormatter migration guide: Migrate to System.Text.Json (JSON)"
+description: "Migrate from BinaryFormatter to System.Text.Json for JSON serialization."
+ms.date: 5/31/2024
+no-loc: [BinaryFormatter, Serialization, System.Text.Json]
+dev_langs:
+  - CSharp
+helpviewer_keywords:
+  - "BinaryFormatter"
+  - "System.Text.Json"
+  - "serializing objects"
+  - "serialization"
+  - "objects, serializing"
+---
+
+# Migrate to System.Text.Json (JSON)
+
+[System.Text.Json](../system-text-json/overview.md) defaults to emphasizing literal, deterministic behavior and avoids any guessing or interpretation on the caller's behalf. The library is intentionally designed this way for security and performance. From the migration perspective, it's crucial to know the following facts:
+
+- By default, **fields are not serialized or deserialized**, but they can be [annotated for serialization](../system-text-json/fields.md), or `JsonSerializerOptions.IncludeFields` can be cautiously set to `true` to include all fields.
```

Comment:
The content of this page seems to immediately give guidance that seems to contradict general best practices for System.Text.Json. I believe the spirit of these notes about fields and private members is meant to help folks serialize their types without having to make many changes to them, but I'm not sure that context is spelled out clearly enough.

Perhaps instead of an "it's crucial to know..." approach, we could first suggest that the API shapes be changed to avoid serializing fields and private members, clarifying when the `[JsonInclude]`,  and `JsonSerializerOptions.IncludeFields` features should be used (with caution).

We will want to make sure @eiriktsarpalis signs off on the content of this page, with it balancing a promotion of STJ best practices and the realistic scenario of folks needing the lightest touch migration from BinaryFormatter.

Reply:
My goal for this paragraph was to explain how it works, so users are not surprised that a new serializer is skipping fields which are not exposed by the properties.

IMO we could keep this part, but add step by step part and describe the recommended approach.

Reply:
I find the emphasis appropriate given that the docs are directed to BF users. They are bound to stumble on issues related to member visibility and polymorphism so it's important that they understand the STJ restrictions (and potential workarounds) from the start.